### PR TITLE
Dynamic Dashboard: Most recent orders - Load order statuses from remote

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #12792 

## Description

Loads the order statuses from remote for the "Most recent orders" dashboard card. 

Changes
- Loads orders statuses from remote for the "Most recent orders" dashboard card. 
- Saves and restores status selection.
- Adds unit tests for the card's view model.

Pending work related to this "Most recent orders" dashboard card is tracked at #12792

## Testing instructions

- Create a JN site with custom order statuses. I used this plugin to create orders with custom statuses. https://wordpress.org/plugins/custom-order-statuses-woocommerce/
- Enable the feature flag `dynamicDashboardM2` and build and login into the app.
- Enable the "Most recent orders" card on the dashboard.
- Confirm that the card now displays the 3 most recent orders. If there are no orders available empty view should be displayed.
- Change the status filter and validate the orders are listed based on the selected status. Observe that status picker also displays the custom order status. 

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                .filter({ $0.status == .cancelled })
+                .sorted(by: { $0.dateCreated > $1.dateCreated })
+                .map({ $0.orderID })
+                .prefix(3)
+        )
+        XCTAssertEqual(viewModel.rows.map({ $0.id }), orderIDs)
+    }
+}
+
+extension LastOrdersDashboardCardViewModelTests {
```
Comment (itsmeichigo) [inline WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModelTests.swift]: Super nit:

```suggestion
private extension LastOrdersDashboardCardViewModelTests {
```

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit:

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModelTests.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModelTests.swift]: Fixed in https://github.com/woocommerce/woocommerce-ios/pull/12847/commits/e35623001255d54fcc1035d77f9e090a6f5a92ef

**Pull request merged**